### PR TITLE
docs: Raphtory Deploy

### PR DESCRIPTION
**Pull request by rutujasurve94**
Docs for Raphtory Deploy:

- [x] `RaphtoryService`
- [x] `Raphtory`

### CODE REVIEW

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 1]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  * ## Methods
+  *
+  *   {s}`createGraph(spout: Spout[T] = new IdentitySpout[T](), graphBuilder: GraphBuilder[T], customConfig: Map[String, Any] = Map()): RaphtoryGraph[T]`
+  *    : Create Graph using spout, graph-builder and custom config
+  *
+  *   {s}`createClient(deploymentID: String = "", customConfig: Map[String, Any] = Map())`
+  *    : Create Client
+  *
+  *   {s}`createSpout(spout: Spout[T])`
+  *    : Create Spout
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:39]: Yes this creates a spout, as the name suggests, but what kind of spout class can you give to this? any examples? what does this spout do?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 2]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`createGraphBuilder(builder: GraphBuilder[T])`
+  *    : Create Graph Builder
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:42]: same comment as the spout, any examples of graph builders? what does a graph builder actually do?  where is the graph builder used?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 3]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`createPartitionManager()`
+  *    : Creates Partition Manager
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:45]: what is a parition manager exactly? where is it used? what does it do? what type is returned?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 4]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`createQueryManager()`
+  *    : Creates Query Manager
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:48]: what is a query manager exactly? where is it used? what does it do? what type is returned?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 5]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`getDefaultConfig(customConfig: Map[String, Any] = Map()): Config`
+  *    : Returns default config
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:51]: what is the default config? where is that located? what kind of custom config can we use? can we have an example of what a custom config is?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 6]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`confBuilder(customConfig: Map[String, Any] = Map()): Config`
+  *    : Returns Config
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:54]: the conf builder is pretty important, its used almost everywhere, can we please explain where is the config this returns? how is that generated? what kind of custom config can we use? can we have an example of what a custom config is?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 7]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
+  *
+  *   {s}`createSpoutExecutor(spout: Spout[T], conf: Config, pulsarController: PulsarController): SpoutExecutor[T]]`
+  *    : Create spout executor
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:57]: what is a spout executor? what does it do?

File: src/main/scala/com/raphtory/core/deploy/Raphtory.scala [hunk 8]
```diff
@@ -23,6 +23,53 @@ import org.apache.pulsar.client.api.Schema
...
 import scala.reflect.runtime.universe._
 
+/**
+  * {s}`Raphtory`
+  *  : `Raphtory` object for creating Raphtory Components
+  *
+  * ## Methods
+  *
+  *   {s}`createGraph(spout: Spout[T] = new IdentitySpout[T](), graphBuilder: GraphBuilder[T], customConfig: Map[String, Any] = Map()): RaphtoryGraph[T]`
+  *    : Create Graph using spout, graph-builder and custom config
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/Raphtory.scala:33]: What kind of object does this return? what can we do with that object? 
What goes in the custom config are there any examples? what can we do with this client object that this creates?

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 9]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
 import scala.reflect.ClassTag
 import scala.reflect.runtime.universe.TypeTag
 
+/**
+  * {s}`RaphtoryService[T]`
+  *
+  *  : `RaphtoryService` for initialising spout, graphbuilder, deploying spout, partition and query manager
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:15]: this is actually an abstract class, that has its some of its functions implemented later on within the code base, this doc should talk about this, what does this service do? examples of where it is used?  alot of our examples extend this class and add functionality, its one of the core components, the current description does not mention this

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 10]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  * ## Methods
+  *
+  *    {s}`defineSpout(): Spout[T]`
+  *      : Initialise Spout
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:20]: this is an abstract method, any examples of how it is overridden when the class is extended? for example you could say the defineSpout is a function that is overriden by objects that extend the raphtory service, this is used to construct the spout which is used to ingest data. an example of ovveriding this fn is as follows. e.g.

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 11]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  *    {s}`defineBuilder(): GraphBuilder[T]`
+  *      : Initialise GraphBuilder
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:23]: see my spout comment

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 12]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  *    {s}`spoutDeploy(): Unit`
+  *      : Deploy spout
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:26]: yes this deploys the spout, but how? what function does it call? where does it get the spout from?

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 13]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  *    {s}`builderDeploy(): Unit`
+  *      : Deploy builder
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:29]: see deploy spout comment

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 14]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  *   {s}`partitionDeploy(): Unit`
+  *      : Deploy partition
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:32]: see deploy spout comment

File: src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala [hunk 15]
```diff
@@ -9,6 +9,46 @@ import org.slf4j.LoggerFactory
...
+  *
+  *   {s}`queryManagerDeploy(): Unit`
+  *      : Deploy query manager
```
Comment (Haaroon) [inline src/main/scala/com/raphtory/core/deploy/RaphtoryService.scala:35]: see deploy spout comment

**Review by Haaroon — CHANGES_REQUESTED** [review]

**Comment by rutujasurve94** [comment]
Thanks a lot for the detailed review @Haaroon , I've updated the documentation accordingly to be very much in depth for ease of onboarding new developers to Raphtory.

**Pull request merged**